Nxos static route track

dmurarasu opened the pull request:
##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
Adding track option for static routes when using nxos_static_route module
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
Fixes #47148 

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Feature Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
nxos_static_route

##### ANSIBLE VERSION
<!--- Paste verbatim output from "ansible --version" between quotes -->
```paste below
ansible 2.8.0.dev0 (devel 5cd1ba3477) last updated 2018/10/06 21:45:11 (GMT +000)
  config file = None
  configured module search path = [u'/home/vagrant/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/vagrant/ansible/lib/ansible
  executable location = bin/ansible
  python version = 2.7.12 (default, Dec  4 2017, 14:50:18) [GCC 5.4.0 20160609]
```

##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->
This PR adds extra network parameter mentioned in #47148 
Tested on 93180YC-EX running NXOS: version 7.0(3)I6(1)
Added documentation and test cases as well.
<!--- Paste verbatim command output below, e.g. before and after your change -->
```paste below

```

kustodian committed:
Fix pip idempotence in check mode

PIP package names must be case insensitive, and must consider hyphens
and underscores to be equivalent
(https://www.python.org/dev/peps/pep-0426/#name), because of this the
module didn't work correctly in check mode. For example if the passed
package name had a different case or an underscore instead of a hyphen
(or the other way around) compared to the installed package, check mode
reported as changed, even though packages were installed. Now the module
ignores case and hyphens/underscores in package names, so check mode
works correctly.

bcoca committed:
avoid spurious warnigns on unique filter

fixes #46189

dmurarasu committed:
Added track option for nxos_static_route module

dmurarasu committed:
added documentation

Commit:
fixed blank lines

Qalthos reviewed (approved):
Looks broadly correct, but I'd like @trishnaguha to have a look as well

mattclay committed:
Fix passing of env vars to Shippable.

trishnaguha commented on lib/ansible/modules/network/nxos/nxos_static_route.py:
```diff
@@ -144,9 +148,30 @@ def remove_route_command(prefix, w):
+def get_configured_track(module, ctrack):
+    check_track = 'Track {0}'.format(ctrack)
+    track_exists = False
+    command = 'show track | awk "/Track/" | no-more'
```
I tried to test your PR. `awk` is not a valid command for my device. It will fail on the devices where `awk` is not supported.
I am on a 7K device. Can you run the command `show track` only and then extract track information?

SamyCoenen committed:
Added gitlab_runner module (#47727)

* Added gitlab_runner module

Commit:
Added nonstring='passthru' when calling to_native to avoid passing a 'None' string to the ipa client (#48157)

larsbu committed:
facts: suse: fix SLES4SAP12 detection (#44388)

Akasurde committed:
ansible-doc: list types of plugin available (#47055)

This fix allows user to view types of plugin available in
ansible-doc command.

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

emonty committed:
Test host in inventory by name in remove_host (#45639)

The host argument is a Host object, and is used as such by
group.remove_host. However, self.hosts is a dictionary of host name to
Host object. Thus, the existing code is checking to see if the Host
object is one of the keys.

Use host.name to interact with the keys of the dictionary.

mmazur committed:
k8s_facts: fix handling of unknown resource types (#47857)

mnecas committed:
ovirt require storage_domain if state is registred (#47423)

DBendit committed:
[docker_network] Add handling for Python booleans in driver_options (#48105)

Fixes #26708

k0ste committed:
mail: fixed STARTTLS module working with python 3.7.0 (#47412)

felixfontein committed:
Add docker_container_facts module. (#48062)

mnecas committed:
add old usages to network cluster (#47185)

* add old usages to network cluster

* set default value to old usages

* rename cluster_network_service to cluster_network_entity

sivel committed:
Add toml inventory plugin (#41593)

* First pass at a toml inventory

* Make EXAMPLES yaml

* Remove unnecessary comment

* Small formatting changes

* Add ansible-inventory option to list as TOML

* TOML inventory improvements, to allow a more simple inventory, specifically related to children

* changelog

* Simplify logic

* Dedupe _expand_hostpattern, making it available to all inventory plugins

* Don't make the TOML inventory dependent on the YAML inventory

* Quote IP address values

* Add more TOML examples

* Further cleanups

* Enable the toml inventory to run by default

* Create toml specific dumper

* 2.8

* Clean up imports

* No toml pygments lexer

* Don't raise an exception early when toml isn't present, and move toml to the end, since it requires an external dep

* Require toml>=0.10.0

* Further clean up of empty data

* Don't require toml>=0.10.0, but prefer it, add code for fallback in older versions

* Ensure we actually pass an encoder to toml.dumps

* Simplify recursive data converter

* Appease tests, since we haven't limited controller testing to 2.7+

* Update docstring for convert_yaml_objects_to_native

* remove outdated catching of AttributeError

* We don't need to catch ImportError when import ansible.plugins.inventory.toml

* Add note about what self.dump_funcs.update is doing

* Address some things

* A little extra comment

* Fix toml availability check

* Don't create an intermediate list

* Require toml file extension

* Add metadata

* Remove TOML docs from intro_inventory to prevent people from getting the wrong idea

* It's in defaults, remove note

* core supported, indicate very clearly that this is preview status

andrewthetechie committed:
Add threading to slack notification module (#47333)

* Add threading to slack notification module

* Fix sanity check

* Change thread_ts to thread_id

JohnDeSilva committed:
Updated Red Hat Repo (#48173)

Updated Red Hat Repo to current version number

chenzhiwei committed:
if we already have stderr, we don't need stderr_lines (#44675)

bcoca committed:
make fact gathering obey play tags (#44717)

fixes #44251

s-hertel committed:
[AWS] Allow the profile to take precedence to allow better compatibility between ~/.boto and ~/.aws/credential files (#45806)

OxHobbs committed:
Ansible Vault and Azure Key Vault vault password script (#44544)

* added new vault password files that can be used with Azure Key Vault

* fixed pylint errors

* fixed pep 8 violations

maxamillion committed:
yum also parse obsolete package output (#45365)

* yum also parse obsolete package output

This is a rebase of the patch originally proposed in
https://github.com/ansible/ansible/pull/40001 by machacekondra

Fixes #39978

Signed-off-by: Adam Miller <admiller@redhat.com>

* properly parse the obsoletes, provide a new output entry, add changelog

Signed-off-by: Adam Miller <admiller@redhat.com>

* make pep8 happy

Signed-off-by: Adam Miller <admiller@redhat.com>

* remove q debugging output

Signed-off-by: Adam Miller <admiller@redhat.com>

kbreit committed:
meraki_vlan - Add note explaining VLAN support tests in a Meraki network (#48278)

* Add note explaining VLAN support tests in a Meraki network

* Small fix.

jborean93 committed:
Added basic equivalent to PowerShell modules (#44705)

* Added basic equivalent to PowerShell modules

* changes based on latest review

* Added tests

* ignore sanity test due to how tests are set up

* Changes to work with PSCore

* Added documentation and change updated more modules

* Add some speed optimisations to AddType

* fix some issues in the doc changes

* doc changes

theo-o committed:
virt: Fix return type of list_vms (#48160)

ilicmilan committed:
Minor fix in one_vm documentation (#48288)

mcassaniti committed:
win_updates: Add post search category matching to support product matching (#45708)

* win_update: Add post search category matching to support product matching

* win_updates: Return categories of each update

* win_updates: Documentation fix-up

* win_updates: Adjusted documentation to reflect regex vs sub-string match of post-cat strings

* win_updates: Sped up post-category checking

* win_updates: Updated documentation to suggest querying post-category strings

* win_updates: Simplified saving and checking post-categories

* fixed some issues and added filtered categories to return value

* win_updates: Moved all category matching to occur after initial search

* win_updates: Adjustments to satisfy PowerShell lint checks

* win_updates: Dropped category validation from action plugin

* win_updates: Documentation updates

* win_updates: Fixed plugin unit tests

nicolasbock committed:
Update os_image.py (#47222)

The description of the `name` and `id` options are updated to highlight the fact that the `name` (which is required and leads to failures if several images exist with the same name) argument can also contain the image `id`.

+label: docsite_pr

bbyhuy committed:
Add survey_spec parameter to module. (#48182)

* Add survey_spec parameter to module.
Fixes #48011

* Removed trailing white space. Added integration test.

RaphaelMeudec committed:
Bug 42787 create volume with label (#46527)

* add None value to docker-version so it can be mocked in tests

njohnston committed:
Typo and style fixes in docs and messages for svc module (#48298)

Minor typo and style fixes in documentation.

Fixed two error messages which incorrectly used "Could" instead of "Could
*not*".

biohazd committed:
Update yum.py (#47749)

add explanation why you must use "shell: yum clean all" and why it won't be included in the yum module as per https://github.com/ansible/ansible/pull/31450#issuecomment-352889579

majekw committed:
Document installation procedure on Slackware (#47657)

Add section with installation procedure on Slackware Linux using sbopkg and SlackBuilds.org build scripts repository.

angystardust committed:
VMware: further improvements to the scenario_clone_template (#48141)

Follow-up to #47889. Fixes the list of required privileges for cloning a vmware template.

bcoca committed:
a more in depth doc on cacheable option (#47878)

* a more indepth doc on cacheable option

bcoca committed:
make inventory warnings a bit smarter (#46284)

less annoying for common cases
 add comment for 'tricky' conditional

mattclay committed:
Get rabbitmq and erlang GPG keys from S3 in tests.

sivel committed:
Support list of dicts with omit. Fixes #45907 (#45923)

Commit:
Do not reboot after updating BIOS configuration

bcoca committed:
clarify insane ini typing (#48169)

* clarify how variables are interpreted in the ini inventory plugin

jeking3 committed:
Improve debug logging of filetree with path being walked and items found (#47884)

mkrizek committed:
conditional: fix direct boolean "shortcut" (#47941)

* conditional: fix direct boolean "shortcut"

* Add unit tests

sivel committed:
Add a new "contains" jinja2 test (#45798)

* Add contains jinja2 test

* backticks

tzurE committed:
IBM_Storage: docs and author changes as requested by Ansible team (#48158)

mkrizek committed:
user: fix removing the expiry time when it's 0 (#47115)

* user: fix removing the expiry time when it's 0

* Improve tests and add changelog

Co-authored-by: Martin Krizek <martin.krizek@gmail.com>

mattclay committed:
Remove ansible version from PR template.

While important for issues, it adds almost no value for PRs.

Akasurde committed:
VMware: Documentation for SSL installation methods (#43992)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

samdoran committed:
Prevent duplicate entries in rhsm_repository module (#48107)

* Complie regular expressions for better performance

* Skip on empty lines

This fixes a bug where the previous repo would be inserted in the result twice since an empty line did not match any of the conditions that would exit the loop iteration.

inxonic committed:
aix_facts: statvfs works on AIX too so make use of it (#46759)

avaika committed:
enchance yaml filter documentation (#46629)

* update to_*_yaml filter documentation

* update to*_yaml filter documentation

shepdelacreme committed:
add changelog fragment for changes merged in #47695 (#48305)

trishnaguha committed:
do not override lldp neighbors nxos_facts (#48087)

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

movinalot committed:
Add UCS DNS Server management module (#46789)

* Add module for UCS DNS Server

movinalot committed:
Add module for UCS Org (#46785)

* Add module for UCS DNS Server

gundalow committed:
basic.py is support:core (#48340)

nijinashok committed:
ovirt_vm: Add option to apply the configuration change on "Next Run" (#48286)

The PR adds new option "next_run" so that user can decide if the
configuration has to be applied on the VM immediately or it has
to be configured on next restart.

Commit:
playbooks_reuse_roles.rst long and shorthand example actually made equivalent. (#48339)

ssiegel committed:
Always use /proc/sys/kernel/random/boot_id to confirm reboot on Linux (#47017)

* Always use /proc/sys/kernel/random/boot_id to confirm reboot on Linux

/proc/sys/kernel/random/boot_id is available since kernel 2.3.16 and
should be safe to rely on.

The previously used method by checking the system boot time using who -b
turned out to be unreliable: Some systems lacking an RTC report the Unix
epoch as boot time, but the code trying to detect that did't always
work.

Closes #46562

* Change DEFAULT_BOOT_TIME_COMMAND

- change to usinsg /proc by default
- add BOOT_TIME_COMMANDS for BSD, Solaris, and macOS

paneu committed:
ios_facts: Strip header from running-config (#48047)

IOS prepends a show running-config with lines that are not part of the
configuration, keeping the output from being an entirely valid configuration:
R1#show run
Building configuration...

Current configuration : 2045 bytes
...

In order to be able to use the config from ios_facts as-is, strip this
header.

machacekondra committed:
Add next_run parameter to ovirt_vms_facts (#48347)

Signed-off-by: Ondra Machacek <omachace@redhat.com>

jimmymccrory committed:
Add run_once as a valid TaskInclude keyword (#48068)

* Add run_once as a valid TaskInclude keyword

* Add changelog fragment

* Add integration test that documents run_once behavior

mkrizek committed:
yum: add integration test for 'update foo*' (#48336)

jacktose committed:
ipaddr filter - return a network address when given an address with /32 subnet (#47539)

* fix #17872

* add changelog fragment and enable unit test

mattclay committed:
Remove unused ops_config action plugin.

dsoper2 committed:
ucs_service_profile_template and integration tests (#48277)

jamescassell committed:
os_server_facts: support all_projects (#44800)

* os_server_facts: support all_projects

* os_server_facts: openstacksdk expects True or False, not None

samdoran committed:
Add setup roles for passlib and pexpect for use with pause and vars_prompt tests (#43613)

* Add passlib to RHEL test instance

This looks like the only tests instance that is missing this libary. It is needed for vars_prompt tests.

* Create setup roles for pexect and passlib

Switch to using aliases rather than installing directly in the test scripts

maxwell-k committed:
Remove unnecessary --recursive option to git clone (#48379)

From installing / Running from source.
Since the repomerge, Ansible doesn't use submodules. Also, between 2.12.0 and 2.13.0 git dropped '--recursive' in favor of '--recurse-submodules'.

sivel committed:
Remove TOML warning on emit (#48385)

sdodsley committed:
Update and correct iSCSI facts collection (#44644)

Co-authored-by: Simon Dodsley <simon@purestorage.com>

gaddman committed:
make password locking in user module idempotent (#43671)

* Simplify logic and add FreeBSD & NetBSD

* Remove incorrect flag for lock and unlock on FreeBSD

* Add tests and changelog


Co-authored-by: Chris Gadd <gaddman@email.com>

Commit:
Fixing issues with azure_rm_virtualmachine crashing (#45269)

* vm fixes

caphrim007 committed:
Fixes for bigip_qkview (#48396)

Fixes unit tests for ansible 2.8 and adds more error handling
for qkview generation

caphrim007 committed:
Removes the f5-sdk from bigip_provision (#48397)

mklassen committed:
Added support for FUNCTION and PROCEDURE privileges to mysql_user (#23223)

mattclay committed:
Remove unused net_config action plugin.

cleming committed:
vultr_server_facts: fix gather facts fails when firewall is enabled on a VM (#48342)

mkrizek committed:
Do not use str() on exceptions (#46950)

FragmentedPacket committed:
Netbox_device.py module (#46936)

* netbox_device module

* Add init.py to each directory

* Fixed a few of the shippable failed tests

* No need for import pynetbox in netbox_utils-removed, changed syntax for set

* A bit more cleanup

* Fixed the 'data' to have suboptions

* Fixed formatting for device_role

* Attempting to fix shippable errors

* Final testing and updated documentation

* Fixed return type and removed testing result files

* Updated some returns to be a list to keep 'meta' formatting consistent

* Updated module to standardize the meta return type

* Updated short_description and added David Gomez as author

* Updated short_description, added David Gomez as author, added module direcotry to BOTMETA.yml

* Updated data type to dict and removed JSON from netbox_utils

Commit:
ec2_instance: ebs_optimized is not sub-option of 'network' (#48341)

* ebs_optimized is not suboption of 'network'

* Add Shaps as ec2_instance maintainer

* Added workaround-backward compatible check for ebs_optimized

* Added ebs_optimized test

* CI fixes, dynamic select of ENA-enabled AMI

resmo committed:
vultr_server_facts: add tests and changelog for firewall group fix in #48342 (#48411)

bcoca committed:
fix tagged implicit gathering (#48294)

* fix tagged implicit gathering

 - None is not tHe default anymore, its an empty list

* Update changelogs/fragments/fix_taggged_gather.yml

Co-Authored-By: bcoca <bcoca@users.noreply.github.com>

remyleone committed:
Add support for Scaleway object storage (#46353)

dsoper2 committed:
support for iscsi vnics based on customer feature request (#48377)

* support for iscsi vnics based on customer feature request
integration tests added for iscsi vnics and vnic state absent

* correct version_added for iscsi and vnic lists

caphrim007 committed:
Adds functional tests for new mysql_user privs (#48403)

Adds tests for FUNCTION and PROCEDURE privs

caphrim007 committed:
Updates to the f5 module utils (#48428)

Updating module utils to align with functionality in modules

acozine committed:
clarifies FAQ answer about env vars (#48432)

larsks committed:
add note regarding {host,group}_vars directory lookup semantics (#48066)

* add note regarding {host,group}_vars directory lookup semantics

While the 'ansible-playbook' command will look for {host,group}_vars in
both the playbook and inventory directories, the 'ansible' command
will only look in the inventory directory. This follows from a close
reading of the documentation but isn't explicit and leads to the
perhaps unexpected result that you can run both commands with the same
configuration in the same directory and get different results.

This commit adds a note to the documentation making the difference in
behavior explicit.

Closes #48065

* incorporates bcoca feedback

caphrim007 committed:
Removes f5-sdk from bigip_pool (#48431)

Additionally fixes forwarding, small bugs, and adds small enhancements

jasonwalsh committed:
Add release argument for redhat_subscription (#47888)

syncpak committed:
cpm_user concat string changes (#45044)

* Add WTI OOB and PDU Device status, control and configuration module

* removed accidental file inclusions

* removed accidental file

* remove unneeded legacy files

* Changed str() to to_native()
Changed + string concat to %

* added to_native() to the remainder of strings being read externally

ematysek committed:
Fix consistency issue in grafana_dashboard module. (#47459)

* Move check for 'dashboard' key to before the create/update if statement.
* Add changelog fragment for PR #47459

jtyr committed:
Support .yaml extension for meta file (#46505)

caphrim007 committed:
Adds new parameters to bigip_pool_member (#48448)

dagwieers committed:
Add msc test targets

caphrim007 committed:
Fixes rebase errors from existing PR. (#48450)

caphrim007 committed:
Remove the f5-sdk from as3 module (#48449)

caphrim007 committed:
Remove f5-sdk from bigip_asm_policy (#48451)

caphrim007 committed:
Minor fixes to bigip_cli_alias (#48452)

caphrim007 committed:
Fixes unit tests that were broken in ansible 2.8 (#48454)

caphrim007 committed:
Removes the f5-sdk from bigip_command (#48455)

caphrim007 committed:
Removes the f5-sdk from bigip_config (#48456)

caphrim007 committed:
Remove the f5-sdk from bigip_configsync_action (#48457)

caphrim007 committed:
Removes f5-sdk from bigip_data_group (#48458)

caphrim007 committed:
Fixes unit tests for bigip_device_auth that were broken in 2.8 (#48459)

caphrim007 committed:
Removes f5-sdk from bigip_device_connectivity (#48460)

caphrim007 committed:
Remove f5-sdk from bigip_device_facts and add more facts (#48462)

caphrim007 committed:
Remove f5-sdk from bigip_device_dns (#48461)

caphrim007 committed:
Remove f5-sdk from bigip_device_group (#48463)

caphrim007 committed:
Removes f5-sdk from bigip_device_group_member (#48464)

sdodsley committed:
Update Pure Storage FlashArray volume module (#45225)

caphrim007 committed:
Fixes ssl cipher ordering in bigip_device_httpd. Also fix unit tests (#48475)

caphrim007 committed:
Removes f5-sdk from bigip_device_license (#48476)

caphrim007 committed:
Remove f5-sdk from bigip_device_ntp (#48477)

caphrim007 committed:
Remove f5-sdk from bigip_device_sshd (#48478)

caphrim007 committed:
Remove f5-sdk from bigip_device_trust (#48479)

caphrim007 committed:
Adds bigip_dns_nameserver module (#48480)

This module can be used to manage DNS nameservers (usually used for
DNS express) functionality on a BIG-IP.

caphrim007 committed:
Removes the f5-sdk from bigip_firewall_address_list module (#48483)

caphrim007 committed:
Fixes unit tests broken in ansible 2.8 and minor module cleanup (#48484)

caphrim007 committed:
Adds bigip_dns_zone module (#48482)

This module can be used to manage zones on a BIG-IP. The managed
zones should **not** be confused with ZoneRunner zones as they are
not the same.

caphrim007 committed:
Adds bigip_firewall_dos_vector module (#48487)

This module can be used to manage DoS vectors in a DoS profile or
device config

caphrim007 committed:
Fixes unit tests for bigip_firewall_policy module (#48490)

caphrim007 committed:
Remove the f5-sdk from bigip_firewall_port_list module (#48492)

caphrim007 committed:
Adds token cleanup and fixes unit tests (#48493)

caphrim007 committed:
Fixes unit tests and move client variable (#48494)

andytom committed:
Add SSL support to postgresql_schema module (#44482)

Have added some extra arguments to the postgresql_schema module to allow
configuring an SSL connection to the postgresql server. The arguments
and method used here for the connection are the same as used by the
postgresql_database module.

caphrim007 committed:
Removes the f5-sdk from bigip_gtm_datacenter (#48495)

caphrim007 committed:
Removes the f5-sdk from bigip_gtm_global (#48496)

caphrim007 committed:
Removes the f5-sdk from bigip_gtm_monitor_bigip (#48497)

caphrim007 committed:
Removes the f5-sdk from bigip_gtm_monitor_external (#48498)

caphrim007 committed:
Remove the f5-sdk from bigip_gtm_monitor_http (#48500)

caphrim007 committed:
Remove f5-sdk from bigip_gtm_monitor_firepass (#48499)

caphrim007 committed:
Remove the f5-sdk from bigip_gtm_monitor_https (#48502)

caphrim007 committed:
Remove f5-sdk from bigip_gtm_monitor_tcp (#48503)

caphrim007 committed:
Remove f5-sdk from bigip_gtm_monitor_tcp_half_open (#48504)

caphrim007 committed:
Remove f5-sdk from bigip_gtm_pool (#48505)

caphrim007 committed:
Remove the f5-sdk from bigip_gtm_pool_member (#48506)

caphrim007 committed:
Remove f5-sdk from bigip_hostname (#48510)

caphrim007 committed:
Add new features to gtm server module (#48507)

Remove the f5-sdk. Fix unit tests. Add new prober parameters

caphrim007 committed:
Remove f5-sdk from bigip_gtm_wide_ip (#48509)

caphrim007 committed:
Remove f5-sdk from bigip_gtm_virtual_server (#48508)

caphrim007 committed:
Remove f5-sdk from bigip_iapp_template (#48512)

caphrim007 committed:
Remove f5-sdk from bigip_iapplx_package (#48513)

caphrim007 committed:
Remove f5-sdk from bigip_iapp_service module (#48511)

caphrim007 committed:
Adds parameters to ike_peer module (#48514)

caphrim007 committed:
Move client variable out to prevent variable declaration errors (#48515)

caphrim007 committed:
Remove f5-sdk from bigip_log_publisher (#48518)

caphrim007 committed:
Remove the f5-sdk from bigip_irule (#48516)

caphrim007 committed:
Remove f5-sdk from bigip_log_destination (#48517)

caphrim007 committed:
Removes f5-sdk from bigip_management_route (#48519)

caphrim007 committed:
Add new parameters to bigip monitor modules (#48520)

caphrim007 committed:
Adds new params to bigip_node. Fix unit tests. (#48521)

caphrim007 committed:
Removes the f5-sdk from bigip_policy (#48523)

caphrim007 committed:
Remove f5-sdk from bigip_policy_rule (#48524)

caphrim007 committed:
Remove f5-sdk from bigip_partition. Fix partition descriptions. (#48522)

caphrim007 committed:
Remove f5-sdk from bigip_profile_dns (#48526)

caphrim007 committed:
Remove f5-sdk from bigip_profile_client_ssl (#48525)

caphrim007 committed:
Adds new parameters to bigip_profile_http (#48528)

caphrim007 committed:
Fixes unit tests in bigip_profile_http_compression (#48529)

caphrim007 committed:
Remove f5-sdk from bigip_profile_oneconnect (#48530)

caphrim007 committed:
Adds bigip_profile_fastl4 module (#48527)

dgamanenko committed:
Fix Deploy VM from template with id==0. (#48421)

caphrim007 committed:
Corrects parameters to flatten correctly. (#48531)

Also fixes unit tests and adds other code patterns that other f5
modules use

caphrim007 committed:
Remove f5-sdk from bigip_profile_tcp module (#48533)

caphrim007 committed:
Remove f5-sdk from bigip_profile_udp module (#48534)

caphrim007 committed:
Adds the bigip_profile_server_ssl module (#48532)

caphrim007 committed:
Adds the bigip_dns_resolver module (#48540)

caphrim007 committed:
Moving client variable out in bigiq modules (#48542)

This is being done to ensure that errors are not encountered if the
exception case is hit

caphrim007 committed:
Removes the f5-sdk from bigip_selfip (#48544)

caphrim007 committed:
Removes the f5-sdk from bigip_routedomain (#48543)

dagwieers committed:
win_psexec: Rewrite using AnsibleModule (#48383)

dagwieers committed:
win_product_facts: Rewrite using AnsibleModule (#48382)

dagwieers committed:
win_wakeonlan: Rewrite using AnsibleModule (#48380)

caphrim007 committed:
Remove f5-sdk from bigip_vcmp_guest (#48541)

dagwieers committed:
win_defrag: Rewrite using AnsibleModule (#48381)

* win_defrag: Rewrite using AnsibleModule

* Fix delta return value

dagwieers committed:
win_tempfile: Rewrite using AnsibleModule (#48386)

* win_tempfile: Rewrite using AnsibleModule

* Fix up error check

* Fix PSParameterizedProperty issue

caphrim007 committed:
Workaround for problems parsing the timezone (#48549)

Device facts gets a time with a timezone back from the builddate.
Unfortunately, python cannot correctly parse timezones by default.
This hacks around the problem.

idahood committed:
Fix LLDP to use json (#48318)

felixfontein committed:
Fixing typo in developing_locally.rst (#48535)

`cacheing` → `caching`

+label: docsite_pr

felixfontein committed:
Docker volume fix labels (#48536)

* Add integration test for labels (which fails).

* Changing labels from list to dict.

* Add changelog.

felixfontein committed:
docker_* modules: updating argument_spec (#48491)

* Updating argument_spec for docker_* modules.

* Adjust docker_network to work with new recursive argument_spec.

* Adjust device IO limits to recursive argument_spec.

* Improve test (test Ansible's cast from str to int).

* Adjust healthcheck options construction.

* Remove superfluous check.

* Make flake8 happy.

* Simplify comparison.

dariko committed:
clarify port.mode paramter requiremets, fail if unmet (#47938)

* clarify port.mode paramter requiremets, fail if unmet

* changelog fragment

* shorten too long line

* remove unnecessary indentation

* test version on docker_version for better maintainability

* normalize imports

* changelog fragment: minor_changes -> bugfixes

* rollback e96a7e57dfefd566fa47cf465a759637affd4795

* typo

Co-Authored-By: dariko <dariko@users.noreply.github.com>

jeking3 committed:
better handling of vmware-vm-shell timeout (#48100)

bcoca committed:
added faq entry on ansible_host + delegation (#48438)

jgkirschbaum committed:
Added short description and moved to rabbitmq subfolder (#48566)

tzurE committed:
IBM storage: fix domain issue by adding more keywords to pyxcli avail… (#48345)

IBM storage: fix domain issue by adding more keywords to pyxcli available fields

bincyber committed:
Updated documentation for the k8s module (#48474)

Qalthos committed:
Match config prompts that aren't maint-mode (#47678)

mkrizek committed:
Add ansible_play_name magic var (#48562)

samdoran committed:
Catch and handle quickly shutdown connections in reboot plugin (#48389)

dagwieers committed:
win_product_facts: Implement review suggestions (#48573)

dagwieers committed:
win_shortcut: Rewrite using AnsibleModule (#48384)

* win_shortcut: Rewrite using AnsibleModule

* Avoid using $args, fix issue

* Small review fixes, and use 'arguments' parameter

* Update doc with new alias version change

nikkytub committed:
Added config-context in hostvars (#47394)

* Added config-contexts in hostvars

* Changed config-contexts to config-context in hostvars

vanyasvl committed:
Fix junos_command.py network module to support spaces in rpc args (#48343)

* Fix junos_command.py network module to support spaces in rpc args

*  Fix junos_command.py network module to support spaces in rpc args

b-t-g committed:
Fix man page template typo. (#48585)

changed from 'direcotry' to 'directory'.

dagwieers committed:
Various small fixes to MSC modules and tests (#48417)

felixfontein committed:
docker_container: make condition less specific (#48590)

* Code seems to have changed from 409 to 400, so let's not check the code.

* Unpause container before removing it.

* Improve code.

* Same for stop_container.

bverschueren committed:
ovirt_vm: fix cloud_init_persist doc (#47854)

* fix cloud_init_persist

* update coud_init_persist doc with default

adarobin committed:
VMware: Add esxi_hostname option to vmware_host_facts (#47062)

Signed-off-by: Adam Robinson <adarobin@umich.edu>

obourdon committed:
Add tests filters (#43221)

This will allow tests to be carried out condtionally if necessary
using regexp include and/or exclude filters
Reorganize imports into alphabetical order for easier insertion

jeking3 committed:
VMware: Add CreateTemporaryDirectoryInGuest capability to vmware_guest_file_operation (#47579)

Ghilli3 committed:
adding fortimanager reviewers (#48581)

* adding fortimanager reviewers

Akasurde committed:
Refactor ingate module_utils (#47959)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

ckotte committed:
VMware: Improve module vmware_local_user_facts (#47868)

* add check mode support
* add assigned role to the output
* change output to match vmware_local_user_manager output
* change principal to user_name
* change full_name to user_description

willthames committed:
Add wait functionality to k8s module (#47493)

Provide wait and wait_timeout parameters and wait for certain
resource kinds to become available.

felixfontein committed:
docker_container: fix tests and idempotency for init and shm_size (#48551)

* Fix tests: use same command if not testing command option.

* Fix idempotency of init option.

* Fix shm_size idempotency (it is included in inspect results from docker API version 1.22 on).

* Add changelog.

dagwieers committed:
Fix author (and test ansibot) (#46060)

Akasurde committed:
ec2_ami_copy: Fix unreferenced variable error (#47598)

Fixes: #47595

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Commit:
Document the tests that are currently run on documentation pull requests. (#48436)

sivel committed:
code-smell test that can find deprecated config items (#46273)

* Start of code-smell test that can find deprecated config items

* Strip deprecated.version from config item

* Don't use os.walk, rely on explictly passed list of files

* Properly disable the test

* Add docs

* Make config an orphan

Mikey032 committed:
VMware: add multiple VLAN ranges support #46150 (#46628)

* add multiple VLAN ranges support
* fix pip8

mscherer committed:
Correct the variable name in example (#48603)

Changed from 'ansilbe_python_interpreter' to 'ansible_python_interpreter'

maxamillion committed:
make yum update_only option actually work (#47942)

* make yum update_only option actually work

Fixes #40615

Signed-off-by: Adam Miller <admiller@redhat.com>

* fix changlog fragment for sanity check

Signed-off-by: Adam Miller <admiller@redhat.com>

* only attempt an update when there are packages to update

Signed-off-by: Adam Miller <admiller@redhat.com>

* fix logic to properly handle the precedent of operations

Signed-off-by: Adam Miller <admiller@redhat.com>

* remove q debug statements

Signed-off-by: Adam Miller <admiller@redhat.com>

jborean93 committed:
powershell - fix async for Ansible.Basic (#48404)

* powershell - fix async for Ansible.Basic

* moved old out outside of the try block

timmc committed:
Clarify what recursive setting of attributes means (#48548)

maxamillion committed:
docs - adapted abadger's changelog announcement email into dev process (#48172)

* docs - adapted abadger's changelog announcement email into dev process

Signed-off-by: Adam Miller <admiller@redhat.com>

* migrate changelog docs to one place, add ref to old location

Signed-off-by: Adam Miller <admiller@redhat.com>

* add changes based on feedback

Signed-off-by: Adam Miller <admiller@redhat.com>

* Refines changelog docs, minor re-org of page

* moves links and anchors for new content

* Further reorg of page

* fixes header per jamescassell

mattclay committed:
Use correct interpreter for ansible-test injector.

mattclay committed:
Relax ansible-test python version checking.

jborean93 committed:
win_shortcut: add run as admin and fix shell folder idempotency (#48584)

mattclay committed:
Add change classification for powershell wrappers.

mkrizek committed:
Run "update foo*" integration test on dnf too (#48607)

jborean93 committed:
win_chocolatey - move over to AnsibleModule and add allow_multiple (#48370)

mattclay committed:
Add test to require docs for action plugins. (#48376)

Commit:
Adding missing idempotence support in load balancer (#45548)

idempotence

Akasurde committed:
Fix typo in example (#48651)

Changed from 'availabily' to 'availability'

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

ckotte committed:
Remove datacenter_name option from vmware_host_datastore (#47824)

The datacenter name isn't required to create a NFS or VMFS datastore.
The option isn't available in the spec.

Akasurde committed:
VMware: Refactor vmware_cluster (#37072)

* Update cluster logic
* Refactor code
* Documentation update

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

mkrizek committed:
apt: disable ubuntu repos to not change test env

justjais committed:
fix nios modules (#48559)

* resolves 47839

* update name

* update name

Signed-off-by: Sumit Jaiswal <sjaiswal@redhat.com>

* update name tests

Signed-off-by: Sumit Jaiswal <sjaiswal@redhat.com>

* update name tests

Signed-off-by: Sumit Jaiswal <sjaiswal@redhat.com>

* modifying the condn

Signed-off-by: Sumit Jaiswal <sjaiswal@redhat.com>

gundalow committed:
module_utils/vmware is support:community (#48627)

mkrizek committed:
Add new meta task end_host (#47194)

* Add new meta task  end_host

TorPeeto committed:
pagerduty_updates (#48367)

Commit:
add N3K-F type to nxos platforms (#47944)

thaumos committed:
Updated roadmap to reflect dates for 2.8 (#48626)

* Updated roadmap to reflect dates for 2.8

ryansb committed:
[AWS] Enable Route53 integration tests (#46228)

Commit:
Added a bytes cast when checking for terminal response (#48427)

abadger committed:
Add an example to serial usage (#48669)

orthanc committed:
[AWS] ses rule set module for inbound email processing (#42781)

* Add module ses_rule_set for Amazon SES

* Update behaviours and naming to be consistent with other aws_ses_ modules.

* Add global lock around tests using active rule sets to prevent intermittent test failures.

* Fix deletion of rule sets so that we don't inactivate the active rule set
when force deleting an inactive rule set.

mattclay committed:
Correct ansible-test injector python behavior.

Inject a symlink to the correct python into the copied injector
directory instead of altering the shebang of the injector. This
has the side-effect of also intercepting `python` for integration
tests which simplifies cases where it needs to be directly invoked
without collecting code coverage.

mattclay committed:
Fix ansible-test merge change detection.

mkrizek committed:
committer_guidelines.rst: update core committers (#48662)

* committer_guidelines.rst: update core committers

* Another update

mattclay committed:
Fix ansible-test interpreter tracking.

Track the interpreter for each copy of the injector by the interpreter
path instead of the interpreter version. This avoids the possibility
of mixing different interpreters with the same version.

pilou- committed:
Remove hekonsek from team_scaleway

and don't notify contributor anymore.

dmurarasu committed:
changed show command to show track

dmurarasu committed:
changed show command to show track

dmurarasu committed:
Merge branch 'nxos_static_route_track' of https://github.com/dmurarasu/ansible into nxos_static_route_track

dmurarasu committed:
Merge branch 'nxos_static_route_track' of https://github.com/dmurarasu/ansible into nxos_static_route_track
